src/unlimited.cpp

```diff
@@ -69,6 +69,14 @@ bool MiningAndExcessiveBlockValidatorRule(const uint64_t newExcessiveBlockSize,
     LOGA("newMiningBlockSize: %d - newExcessiveBlockSize: %d\n", newMiningBlockSize, newExcessiveBlockSize);
     return (newMiningBlockSize <= newExcessiveBlockSize);
 }
+std::string AcceptDepthValidator(const unsigned int &value, unsigned int *item, bool validate)
```

Comment:
nit: why not use C++11 `uint` notation rather than `unsigned int`?

Reply:
The type must match the type of the CTweak which is "unsigned int".  And BTW there is no standard uint type.  There's uint32_t, etc but there is no reason to constrain the # bits in this case to something that may not be the most efficient for the host CPU.

Reply:
with `uint` I was actually referring to one of the incarnations you see to fit better. I should have used `uint*`.

Reply:
for greatest time efficiency you want to use the machine's register size.  Sure you might waste 4 bytes but that is nothing, unless you are dealing with huge arrays.  Its been awhile but let's say we want to load, add and store a 16 bit value on a 32 bit machine:
mov eax, [ebp]  # load register eax with the data at address ebp
mov ecx, eax
and ecx, 0xffff0000  # remember the high bytes
and eax, 0xffff # forget the high bytes in eax in preparation for our add (not strictly necessary)
add eax, ebx
and eax, 0xffff
or eax, ecx  # combine the high bytes with your add
mov [ebp], eax  # store it all back into memory

and the above isn't quite a perfect 16 bit add because it doesn't set the carry bit in the status register if your 16 bit add overflows, but the compiler could detect that it isn't using a jump if carry (jc) instruction anyway so optimize that away.

verses:
mov eax, [ebp]
add eax, ebx
mov [ebp], eax

The intel architecture does allow non-word memory access, and half-register access through "ax, bx, etc" so things would be cleaner than what I wrote above (for intel anyway) but use of these is likely just doing the equivalent of the above in the processor's microcode.  

The above is a good mental model and is the way it works for RISC machines like ARM.  The reality for Intel machines is they just tell you that short-word operations are slower but don't explain what is going on under the covers for every operation.

Reply:
mine was more a stylistic complain/question rather than anything else (i.e. use the C++11 notation when possible), both time and space efficiency are not crucial in this case as you explained so well.

Reply:
hmm... there are some c++11 things that are "better" and replace c or c++ stuff and other things just augment.  I don't see the bit-specified integers as replacing or always better the c.  I believe in the concept of minimal specification.  So if the integer size doesn't matter (within the c definition) then I'd advocate not specifying it.  Otherwise you have engineers thinking about why 16, 32 or 64 bits are so important in this situation...